framework/include/cppmicroservices/detail/BundleTracker.tpp

```diff
@@ -0,0 +1,273 @@
+/*=============================================================================
+
+  Library: CppMicroServices
+
+  Copyright (c) The CppMicroServices developers. See the COPYRIGHT
+  file at the top-level directory of this distribution and at
+  https://github.com/CppMicroServices/CppMicroServices/COPYRIGHT .
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+=============================================================================*/
+
+#include "cppmicroservices/Bundle.h"
+#include "cppmicroservices/BundleContext.h"
+
+#include "cppmicroservices/detail/BundleTrackerPrivate.h"
+#include "cppmicroservices/detail/TrackedBundle.h"
+
+#include <optional>
+#include <vector>
+#include <type_traits>
+
+namespace cppmicroservices {
+
+template<class T>
+template <typename S0, typename... S>
+constexpr typename BundleTracker<T>::BundleState BundleTracker<T>::CreateStateMask(S0 const &s0, S const &...s) {
+  static_assert((std::is_enum_v<S0> && ... && std::is_enum_v<S>),
+                "The function requires enumerations.");
+  static_assert((std::is_same_v<Bundle::State, std::decay_t<S0>> && ... &&
+                 std::is_same_v<Bundle::State, std::decay_t<S>>),
+                "All values must be Bundle States.");
+  return (s0 | ... | s);
+}
```

Comment:
Mentioned earlier. Is it possible to remove the template type `T` dependency here as it is not needed?

Reply:
Moved to header